fedpkg: remove unneeded patch

marsam opened the pull request:
###### Motivation for this change
This patch is no longer used

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

marsam committed:
fedpkg: fix patch

jtojnar commented:
@GrahamcOfBorg build python3.pkgs.fedpkg python2.pkgs.fedpkg

GrahamcOfBorg commented:
<!--REQUEST_ID=6e773a2b-b634-4fa8-ab97-f51b963e71c1-->
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.51229&attempt_id=f90103b2-139f-467d-b72b-0d25acd17093)

The following builds were skipped because they don't evaluate on x86_64-darwin: python3.pkgs.fedpkg, python2.pkgs.fedpkg

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=6e773a2b-b634-4fa8-ab97-f51b963e71c1-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.51229&attempt_id=0b0ae8b3-c574-45df-9649-843ea6874748)

Attempted: python2.pkgs.fedpkg

The following builds were skipped because they don't evaluate on x86_64-linux: python3.pkgs.fedpkg

<details><summary>Partial log (click to expand)</summary><p>

```
/build/fedpkg-1.29
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/cyphpr9xrrzij259jb12s8as6bsi0h5r-python2.7-fedpkg-1.29
strip is /nix/store/n4hb93w6j076xcjw5pm09rdmc09s075b-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/cyphpr9xrrzij259jb12s8as6bsi0h5r-python2.7-fedpkg-1.29/lib  /nix/store/cyphpr9xrrzij259jb12s8as6bsi0h5r-python2.7-fedpkg-1.29/bin
patching script interpreter paths in /nix/store/cyphpr9xrrzij259jb12s8as6bsi0h5r-python2.7-fedpkg-1.29
checking for references to /build in /nix/store/cyphpr9xrrzij259jb12s8as6bsi0h5r-python2.7-fedpkg-1.29...
wrapping `/nix/store/cyphpr9xrrzij259jb12s8as6bsi0h5r-python2.7-fedpkg-1.29/bin/fedpkg'...
wrapping `/nix/store/cyphpr9xrrzij259jb12s8as6bsi0h5r-python2.7-fedpkg-1.29/bin/fedpkg-stage'...
/nix/store/cyphpr9xrrzij259jb12s8as6bsi0h5r-python2.7-fedpkg-1.29
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=6e773a2b-b634-4fa8-ab97-f51b963e71c1-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.51229&attempt_id=62e93d47-23a8-455a-92da-5b615b18b40f)

Attempted: python2.pkgs.fedpkg

The following builds were skipped because they don't evaluate on aarch64-linux: python3.pkgs.fedpkg

<details><summary>Partial log (click to expand)</summary><p>

```
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/sy1gpmjq3apymrvraz42a29rpy9n46hm-python2.7-fedpkg-1.29
strip is /nix/store/qjrnv0qw44bw1hc23zhfh26xd1c25dfs-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/sy1gpmjq3apymrvraz42a29rpy9n46hm-python2.7-fedpkg-1.29/lib  /nix/store/sy1gpmjq3apymrvraz42a29rpy9n46hm-python2.7-fedpkg-1.29/bin
patching script interpreter paths in /nix/store/sy1gpmjq3apymrvraz42a29rpy9n46hm-python2.7-fedpkg-1.29
checking for references to /build in /nix/store/sy1gpmjq3apymrvraz42a29rpy9n46hm-python2.7-fedpkg-1.29...
wrapping `/nix/store/sy1gpmjq3apymrvraz42a29rpy9n46hm-python2.7-fedpkg-1.29/bin/fedpkg'...
wrapping `/nix/store/sy1gpmjq3apymrvraz42a29rpy9n46hm-python2.7-fedpkg-1.29/bin/fedpkg-stage'...
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy
/nix/store/sy1gpmjq3apymrvraz42a29rpy9n46hm-python2.7-fedpkg-1.29
```
</p></details>

worldofpeace commented:
@marsam Can you remove `name = "${pname}-${version}";`? 
We don't need that in python packages anymore.

jtojnar commented:
The patch might still be needed:

```
$ exa -T result/
result
├── bin
│  ├── fedpkg
│  └── fedpkg-stage
├── lib
│  └── python2.7
│     └── site-packages
│        ├── etc
│        │  ├── bash_completion.d
│        │  │  └── fedpkg.bash
│        │  └── rpkg
│        │     ├── fedpkg-stage.conf
│        │     └── fedpkg.conf
│        ├── fedpkg
│        │  ├── __init__.py
│        │  ├── __init__.pyc
│        │  ├── __main__.py
│        │  ├── __main__.pyc
│        │  ├── cli.py
│        │  ├── cli.pyc
│        │  ├── lookaside.py
│        │  └── lookaside.pyc
│        ├── fedpkg-1.29.dist-info
│        │  ├── COPYING
│        │  ├── INSTALLER
│        │  ├── METADATA
│        │  ├── RECORD
│        │  ├── top_level.txt
│        │  └── WHEEL
│        └── usr
│           └── share
│              └── zsh
│                 └── site-functions
│                    └── _fedpkg
└── nix-support
   └── propagated-build-inputs
```

worldofpeace commented:
Perhaps an update and a slightly different patch would be preferable https://pagure.io/fedpkg/blob/master/f/setup.py#_51

jtojnar commented on pkgs/development/python-modules/fedpkg/fedpkg-buildfix.diff:
```diff
@@ -0,0 +1,17 @@
+diff -ruN orig/setup.py new/setup.py
```
Could you remove this line and the dates from the next two? It will make for less changes when the patch is updated. Also the file does not fix build, so `fix-paths.patch` would be a better name.

marsam replied:
sure, updated

worldofpeace commented:
@marsam Would you mind doing other cleanups for this?

marsam commented:
sure, do you have anything in mind?

worldofpeace commented:
> sure, do you have anything in mind?

Yep, just wanted to ask before I started reviewing again.

worldofpeace reviewed (commented):
I think this a tool and not meant to be used as an importable module.

So we should use `buildPythonApplication` (similar to what was done in #50863) and move it to a better home in nixpkgs.

worldofpeace commented:
Also, all that cleanup I think an update would be cool.

marsam committed:
fedpkg: promote to python application

marsam committed:
fedpkg: add fedora_cert as dependency

[1: 1b9fd365819] Removed fedora_cert from nixpkgs, but fedpkg 1.29 still
it required at runtime.

1: 1b9fd36581951b8a8c51e6e1cc792c1589135717
   fedora_cert: remove package

worldofpeace commented:
>  I've updated fedpkg with buildPythonApplication, and checked the binaries are working

:+1: 

> Also, I did try to update it, but that it's going to take a bit of time: 3c1823c, because they don't quite follow the standard python packaging. I think that can go in another PR.

I agree that a separate pr for that would be appropriate.

worldofpeace reviewed (approved):
Builds and executes locally. Also I can clone a repo anonymously.
Though I had to specify the path to a modified config with the new urls.
